### PR TITLE
feat: flag interaction

### PR DESCRIPTION
**Pull request by denis-gandel**
# Pull Request

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality
      to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: src/entities/concretes/elements/touchable/flag.py [hunk 1]
```diff
@@ -14,6 +37,42 @@ def __init__(
             position,
             elements[ElementType.FLAG][element_sub_type],
             x_rect_percent=(
-                0.11 if element_sub_type == ElementSubType.FLAG_SUPPORT else 1
+                0.145 if element_sub_type == ElementSubType.FLAG_SUPPORT else 1
             ),
         )
+        self.__points = self.__get_points()
+
+    def __get_points(self) -> int:
```
Comment (daniel-espinoza-79) [inline src/entities/concretes/elements/touchable/flag.py:45]: Chaining of conditionals can be avoided if constants are handled in lists

by example:

RANGES = [
        (240, 299),
        (300, 359),
        (360, 419),
        (420, 539),
        (540, 659),
        (660, float('inf'))  # Usamos 'inf' para indicar que no hay límite superior
    ]
    
    SCORES = [
        2000,  # SCORE_FIRST_RANGE
        1000,  # SCORE_SECOND_RANGE
        800,   # SCORE_THIRD_RANGE
        400,   # SCORE_FOURTH_RANGE
        200,   # SCORE_FIFTH_RANGE
        100    # SCORE_FINAL_RANGE
    ]

File: src/entities/concretes/elements/touchable/flag.py [hunk 2]
```diff
@@ -14,6 +37,42 @@ def __init__(
...
+        elif self.SECOND_RANGE[0] <= y_position <= self.SECOND_RANGE[1]:
+            return self.SCORE_SECOND_RANGE
+        elif self.THIRD_RANGE[0] <= y_position <= self.THIRD_RANGE[1]:
+            return self.SCORE_THIRD_RANGE
+        elif self.FOURTH_RANGE[0] <= y_position <= self.FOURTH_RANGE[1]:
+            return self.SCORE_FOURTH_RANGE
+        elif self.FIFTH_RANGE[0] <= y_position <= self.FIFTH_RANGE[1]:
+            return self.SCORE_FIFTH_RANGE
+        elif self.FINAL_RANGE <= y_position:
+            return self.SCORE_FINAL_RANGE
```
Comment (daniel-espinoza-79) [inline src/entities/concretes/elements/touchable/flag.py]: Chaining of conditionals can be avoided if constants are handled in lists

by example:
```python

RANGES = [
        (240, 299),
        (300, 359),
        (360, 419),
        (420, 539),
        (540, 659),
        (660, float('inf'))  # inf is used to indicate that there is no upper limit    
        ]
    
    SCORES = [
        2000,  # SCORE_FIRST_RANGE
        1000,  # SCORE_SECOND_RANGE
        800,   # SCORE_THIRD_RANGE
        400,   # SCORE_FOURTH_RANGE
        200,   # SCORE_FIFTH_RANGE
        100    # SCORE_FINAL_RANGE
    ]
    
     DEFAULT_SCORE = 5000
`

**Review by daniel-espinoza-79 — COMMENTED** [review]
https://github.com/software-development-v/mario-pygame/assets/135346738/e4d26bd1-428d-48b1-b925-d9da2e1ea1d4




there is a bug , when cumpa walks to the flag

**Review by MorveN11 — COMMENTED** [review]
Good Job @denis-gandel

**Review by MorveN11 — APPROVED** [review]

**Pull request merged**